Disable reconsidermostworkchain during initial bootstrap when chain is not synced

sickpig opened the pull request:
This is for testing purposeonly and hence it is disabled by default.
This tweak will be useful during multiple clients interop network upgrade tests.
During this tests  official testnet is forked via invalidate block, that means that if for what ever reason you need to restart your client during the test, you need to rollbackchain and then reconsiderblock the first block of the forked testnet. This is because if more than 1 block at time have to be invalidated so that the utxo may get undone correctly.

ptschip commented on src/globals.cpp:
```diff
@@ -508,6 +508,18 @@ CTweak<double> dMinLimiterTxFee("minlimitertxfee",
+/** Disable reconsidermostworkchain during initial bootstrap when chain is not synced.
+  * This is for testing purpose only and hence it is disabled by default.
+  * This tweak will be useful during multiple clients interop network upgrade tests.
+  * During this tests  official testnet is forked via invalidate block, that means that
```
-During this tests official
+During these tests the official

sickpig replied:
will fix

ptschip commented on src/init.cpp:
```diff
@@ -1357,7 +1357,7 @@ bool AppInit2(Config &config, thread_group &threadGroup)
     // node before a hardfork.
-    if (!fReindex)
+    if (!fReindex || !(avoidReconsiderMostWorkChain.Value()))
     {
```
?

-||
+&&

sickpig replied:
you are right, got that change locally but I forgot to add it :/ Thanks for catching it.

sickpig commented:
rebased

sickpig committed:
Disable reconsidermostworkchain during initial bootstrap when chain is not synced

This is for testing purposeonly and hence it is disabled by default.
This tweak will be useful during multiple clients interop network upgrade tests.
During this tests  official testnet is forked via invalidate block, that means that
if for what ever reason you need to restart your client during the test, you need to
rollbackchain and then reconsiderblock the first block of the forked testnet. This is because
if more than 1 block at time have to be invalidated so that the utxo may get undone correctly.
